### DISCUSSION TITLE
Hosting the docker container in Kubernetes without wiping the db

### DISCUSSION THREAD

**Discussion by mewid** [middlewarehq/middleware#378] [Q&A]
Hello, I'm currently looking into hosting the middleware docker container in our kubernetes cluster. The quickest way to do this, I think, is to download your latest public docker image, and host the container that way.

The obvious problem with that is that the internal postgres database will be reset when the container is taken down and created again.

It would be nice to not have to fork the project to make changes within it to fit our purposes, so im looking into either:
1. Hosting persistant volumes in our kubernetes cluster, or;
2. Hijacking the postgres connection host and port to point to our own postgres database.

I have tried the second approach, but it seems like the db initialisation scripts are relying on the local postgres machine, rather than using the DB_HOST and DB_PORT, so maybe that is a dead end (if we do not run the migrations on our own db first).

Do you have any thoughts and/or suggestions regarding how to host this? (Im not docker or kubernetes expert, so I might be missing obvious things). I do know that you are providing a paid plan to host the solution yourself, but for now, we are at the stage to try this out first and see how useful we find it :)

**Comment by jayantbh** [comment]
> rather than using the DB_HOST and DB_PORT, so maybe that is a dead end (if we do not run the migrations on our own db first).

Looks like this should have worked, and that not working for you might be buggy behaviour.
**We'll look into this asap.**

The goal is that you should totally be able to bring your own postgres instance to run Middleware.

**Reply by mewid** [reply]
I might have misdiagnosed the issue. Im not sure it's because of the initialisation script anymore. It simply cannot find my postgres container. Im not sure why, but I suspect it is because the container is using its hosts DNS, and thus cannot resolve other containers by container name.

Why it's using its host DNS im not sure, but my postgres docker container is also doing that, so maybe it's something on my end.

I might be too early to diagnose this again, but I thought to mention it in case you would put effort into trying to look into it

**Reply by jayantbh** [reply]
Oh yeah we did look into it and weren't sure what's wrong because the setup and configs seemed fine.

However your reply might still enable us to figure out what's not working here.
